spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/EnvironmentController.java

```diff
@@ -72,34 +68,42 @@ public Environment labelled(@PathVariable String name, @PathVariable String prof
 	@RequestMapping("/{label}/{name}-{profiles}.properties")
 	public ResponseEntity<String> labelledProperties(@PathVariable String name,
 			@PathVariable String profiles, @PathVariable String label) throws IOException {
-		if (name.contains("-") || profiles.contains("-")) {
-			throw new IllegalArgumentException(
-					"Properties output not supported for name or profiles containing hyphens");
-		}
+		validateNameAndProfiles(name, profiles);
 		Map<String, Object> properties = convertToProperties(labelled(name, profiles,
 				label));
-		return getSuccess(sortLines(properties));
+		return getSuccess(getPropertiesString(properties));
 	}
 
-	private String sortLines(Map<String, Object> properties) throws IOException {
-		List<String> list = new ArrayList<String>();
-		for (Entry<String, Object> entry : properties.entrySet()) {
-			if (entry.getKey().equals("spring.profiles")) {
-				continue;
-			}
-			String line = entry.getKey() + ": " + entry.getValue();
-			list.add(line);
-		}
-		Collections.sort(list);
```

Comment:
I changed from LinkedHashMap to TreeMap in convertToProperties method.
TreeMap implements SortedMap

This way, the map is sorted by key upon insertion and the order propagates to both the json and the other endpoints.